hack/e2e/README.md

```diff
@@ -0,0 +1,50 @@
+# Eventing End-to-End Tests
+Tests the end-to-end flow for eventing-manager.
+
+## Overview
+
+This test covers the end-to-end flow for Eventing. It is divided in three parts:
+1. `setup` - prepares an Eventing CR and verifies that all the required resources are being provisioned by the eventing-manager.
+2. `eventing` - prepares Subscription CR(s) and tests the end-to-end delivery of events for different event types.
+3. `cleanup` - removes the test resources and namespaces from the cluster.
+
+## Usage
+
+### Pre-requisites
+
+- [Go](https://go.dev/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/)
+- Access to Kubernetes cluster ([k3d](https://k3d.io/) / k8s)
+- Eventing-manager deployed to the cluster without any Eventing CR.
+- For EventMesh backend, the following secrets should exist in the cluster:
+  - Event Mesh credentials secret by name: `eventing-backend` in namespace: `kyma-system`.
+  - OAuth client secret by name: `eventing-webhook-auth` in namespace: `kyma-system`.
+
+### How to run end-to-end tests
+
+1. Prepare the `.env` file based on the `.env.template`.
+
+```
+KUBECONFIG=                  # Kubernetes cluster kubeconfig file path
+BACKEND_TYPE=NATS            # NATS or EventMesh
+MANAGER_IMAGE=               # [Optional] Container image of eventing-manager
+EVENTMESH_NAMESPACE=         # [Optional] Default is: "/default/sap.kyma/tunas-develop"
+```
+
+2. Run the following command to set up the environment variables in your system:
+```bash
```

Comment:
In the beginning of each line of the code block, add 3 blank spaces for proper indentation below the list item.